ui/components/app/confirmations/alerts/alert-modal/alert-modal.tsx

```diff
@@ -225,20 +275,37 @@ export function AlertModal({
           borderWidth={1}
           display={headerStartAccessory ? Display.InlineFlex : Display.Block}
         />
-        <AlertHeader selectedAlert={selectedAlert} />
+        <AlertHeader
+          selectedAlert={selectedAlert}
+          customAlertTitle={customAlertTitle}
+        />
         <ModalBody>
-          <AlertDetails selectedAlert={selectedAlert} />
-          <AcknowledgeCheckbox
+          <AlertDetails
             selectedAlert={selectedAlert}
-            isConfirmed={isConfirmed}
-            setAlertConfirmed={setAlertConfirmed}
+            customAlertDetails={customAlertDetails}
           />
+          {customAcknowledgeCheckbox ?? (
+            <AcknowledgeCheckbox
+              selectedAlert={selectedAlert}
+              isConfirmed={isConfirmed}
+              setAlertConfirmed={setAlertConfirmed}
+            />
+          )}
```

Comment:
rather than adding new props to pass child components to override existing components, I think it'd be better to abstract the logic so that shared logic resides in AlertModal or could be used by AlertModal. Shared logic could also be separate, reusable components.

The labels have the same values so we can remove `confirmAlertModalAcknowledge` and reuse the existing `alertModalAcknowledge`

The customAcknowledgeCheckbox is very similar to the existing one in AlertModal. We could move the existing `AcknowledgeCheckbox` outside of this component and into its own file to be reused. If all AlertModals are guaranteed to have these AcknowledgeCheckbox, we could consider not passing customAcknowledgeCheckbox, but more specific params

similar comments for the other "custom" child components

Reply:
Great point, `AcknowledgeCheckbox` is a good candidate to abstract the logic.
I created an `AcknowledgeCheckboxBase` component to abstract the logic and re-utilized it into the `confirm-alert-modal.tsx`.
Unfortunately, some of those internal components changed over the most recent PRs such as https://github.com/MetaMask/metamask-extension/pull/24214 and https://github.com/MetaMask/metamask-extension/pull/24400. I will reassess the `customAcknowledgeButton` in the PRs mentioned to see if we can do something similar.